File: BAIST/BAIS3150/StudentDemo/App_Code/BAIS3150CodeSampleHandler.vb
Imports Microsoft.VisualBasic

Public Class BAIS3150CodeSampleHandler

    Public Function CreateProgram(ProgramCode As String, Description As String) As Boolean

        Try

            Dim ProgramManager = New Programs()
            Dim Confirmation As Boolean = ProgramManager.AddProgram(ProgramCode, Description)
            Return Confirmation

        Catch e As Exception
            Return False
        End Try

    End Function

    Public Function EnrollStudent(AcceptedStudent As Student, ProgramCode As String) As Boolean

        Try

            Dim StudentManager = New Students()
            Dim Confirmation = StudentManager.AddStudent(AcceptedStudent, ProgramCode)
            Return Confirmation

        Catch ex As Exception
            Return False
        End Try

    End Function

    Public Function GetStudent(StudentID As String) As Student

        Dim StudentManager = New Students()
        Dim EnrolledStudent = StudentManager.GetStudent(StudentID)
        Return EnrolledStudent

    End Function

	Function UpdateStudent(EnrolledStudent As Object) As Boolean

		Dim StudentManager = New Students()
		Dim Confirmation = StudentManager.UpdateStudent(EnrolledStudent)
		Return Confirmation

	End Function

	Function RemoveStudent(StudentID As String) As Boolean

		Dim StudentManager = New Students()
		Dim Confirmation = StudentManager.DeleteStudent(StudentID)
		Return Confirmation

	End Function

End Class

File: BAIST/BAIS3150/ababcock1BAIS3150CodeSampleSite/App_Code/BAIS3150CodeSampleHandler.cs
using Microsoft.VisualBasic;
using System;

public class BAIS3150CodeSampleHandler
{
    public bool CreateProgram(string ProgramCode, string Description)
	{
		try
		{
            var ProgramManager = new Programs();
            bool Confirmation = ProgramManager.AddProgram(ProgramCode, Description);
            return Confirmation;
		}
        catch (Exception e)
		{
			return false;
		}
	}

	public Program FindProgram(string ProgramCode)
	{
		var ProgramManager = new Programs();
		var ActiveProgram = ProgramManager.GetProgram(ProgramCode);
		return ActiveProgram;
	}

    public bool EnrollStudent(Student AcceptedStudent, string ProgramCode)
	{
		try
		{
			var StudentManager = new Students();
            var Confirmation = StudentManager.AddStudent(AcceptedStudent, ProgramCode);
            return Confirmation;
		}
        catch (Exception ex)
		{
            return false;
		}
	}

    public Student FindStudent(string StudentID)
	{
        var StudentManager = new Students();
        var EnrolledStudent = StudentManager.GetStudent(StudentID);
        return EnrolledStudent;
	}

	public bool UpdateStudent(Student EnrolledStudent)
	{
		var StudentManager = new Students();
		var Confirmation = StudentManager.UpdateStudent(EnrolledStudent);
		return Confirmation;
	}

	public bool RemoveStudent(string StudentID)
	{
		var StudentManager = new Students();
		var Confirmation = StudentManager.DeleteStudent(StudentID);
		return Confirmation;
	}
}
